lib/logstash/outputs/s3.rb

```diff
@@ -180,6 +189,11 @@ def create_temporary_file
       end
 
       @tempfile = File.open(filename, "a")
+      
+      #  header row processing
+      if @header_row != "" 
+        @tempfile.syswrite(@header_row + "\n")
+      end
```

Comment:
Correct, once per file and only as the first line.  I'll add a check for filesize == 0 or something equivalent.

Will add a test or two, thanks for the reminder.